OTKBasicVideoCapturer.m

```diff
@@ -0,0 +1,273 @@
+    outputDevice.videoSettings = @{
+                                   (NSString *)kCVPixelBufferPixelFormatTypeKey:@(kCVPixelFormatType_420YpCbCr8BiPlanarVideoRange)
+                                   };
+
+    [outputDevice setSampleBufferDelegate:self queue:self.captureQueue];
+
+    [self.captureSession addOutput:outputDevice];
+
+    // Set framerate
+    double bestFrameRate = [self bestFrameRateForDevice];
+
+    CMTime desiredMinFrameDuration = CMTimeMake(1, bestFrameRate);
+    CMTime desiredMaxFrameDuration = CMTimeMake(1, bestFrameRate);
+
+    [self.inputDevice.device lockForConfiguration:&error];
+    self.inputDevice.device.activeVideoMaxFrameDuration = desiredMaxFrameDuration;
+    self.inputDevice.device.activeVideoMinFrameDuration = desiredMinFrameDuration;
+
+    [self.captureSession commitConfiguration];
+
+    self.format = [OTVideoFormat videoFormatNV12WithWidth: self.outputWidth
+                                                   height: self.outputHeight];
+}
+
+- (void)releaseCapture
+{
+    self.format = nil;
+}
+
+- (int32_t)startCapture
+{
+    self.captureStarted = YES;
+    [self.captureSession startRunning];
+
+    return 0;
+}
+
+- (int32_t)stopCapture
+{
+    self.captureStarted = NO;
+    [self.captureSession stopRunning];
+    return 0;
+}
+
+- (BOOL)isCaptureStarted
+{
+    return self.captureStarted;
+}
+
+- (int32_t)captureSettings:(OTVideoFormat*)videoFormat
+{
+    videoFormat.pixelFormat = OTPixelFormatNV12;
+    videoFormat.imageWidth = self.inputWidth;
+    videoFormat.imageHeight = self.inputHeight;
+    return 0;
+}
+
+#pragma mark - AVCaptureVideoDataOutputSampleBufferDelegate
+
+- (void)captureOutput:(AVCaptureOutput *)captureOutput
+  didDropSampleBuffer:(CMSampleBufferRef)sampleBuffer
+       fromConnection:(AVCaptureConnection *)connection
+{
+    //NSLog(@"Frame dropped");
+}
+
+- (void)captureOutput:(AVCaptureOutput *)captureOutput
+didOutputSampleBuffer:(CMSampleBufferRef)sampleBuffer
+       fromConnection:(AVCaptureConnection *)connection
+{
+    if (!self.captureStarted)
+        return;
+
+    int cropHeight = self.outputHeight, cropWidth = self.outputWidth;
+
+    CVImageBufferRef imageBuffer = CMSampleBufferGetImageBuffer(sampleBuffer);
+    OTVideoFrame *frame = [[OTVideoFrame alloc] initWithFormat:self.format];
+
+    NSUInteger planeCount = CVPixelBufferGetPlaneCount(imageBuffer);
+
+//    uint8_t *buffer = malloc(sizeof(uint8_t) * CVPixelBufferGetDataSize(imageBuffer));
+    uint8_t *buffer = malloc(sizeof(uint8_t) * cropWidth * cropHeight * 1.5f ); //12 = NVI, but take it from other place
+    uint8_t *dst = buffer;
+    uint8_t *planes[planeCount];
+    uint8_t *rowBaseAddress;
+
+    CVPixelBufferLockBaseAddress(imageBuffer, 0);
+    for (int i = 0; i < planeCount; i++) {
+        // Account for UV plane.
+        double uvReduction = pow(2, -i);
+        cropHeight = (int) (cropHeight * uvReduction);
+        cropWidth = (int) (cropWidth * uvReduction);
```

Comment:
I added this when I thought this could be multiple planes, might be more explicit and simpler as:
```objc
if (plane == 1) {
  cropHeight /= 2;
  cropWidth /= 2;
}
```
(Assuming /= works in objc)